### PR TITLE
GSAGH-508 Update the GsaGH code for the load panel to work

### PR DESCRIPTION
**Pull request by SandeepArup**

### CODE REVIEW

File: GsaGH/Components/6_Display/Contour3dResults.cs [hunk 1]
```diff
@@ -541,7 +541,7 @@ private enum FoldMode {
       verticies.AsParallel().AsOrdered();
 
       Parallel.ForEach(elems.Keys, key => {
-        Element element = elems[key];
+        GSAElement element = new GSAElement(elems[key]);
```
Comment (DominikaLos) [inline GsaGH/Components/6_Display/Contour3dResults.cs:544]: use var

File: GsaGH/Helpers/GsaAPI/GsaElement.cs [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+using System.Runtime.Remoting.Lifetime;
+namespace GsaAPI {
+  public enum GSAElementType {
+
+  }
+
+  public class GSAElement {
+    public static int LoadPanelType = -1000;
+    private GSAElement() { }
+    public GSAElement(Element element) { Element = element; IsLoadPanel = false; }
```
Comment (DominikaLos) [inline GsaGH/Helpers/GsaAPI/GsaElement.cs:22]: invalid formatting

**Review by DominikaLos — COMMENTED** [review]
in GsaGH we are using this formatting, pls read: https://google.github.io/styleguide/csharp-style.html

File: GsaGH/Components/6_Display/Contour3dResults.cs (hunk 1 above)
Comment (SandeepArup) [reply GsaGH/Components/6_Display/Contour3dResults.cs]: done

File: GsaGH/Helpers/GsaAPI/GsaElement.cs (hunk 2 above)
Comment (SandeepArup) [reply GsaGH/Helpers/GsaAPI/GsaElement.cs]: done

File: GsaGH/Helpers/GH/RhinoConversions.cs [hunk 3]
```diff
@@ -476,15 +479,16 @@ public class RhinoConversions {
...
+      Mesh mesh, bool isLoadPanel, bool createQuadraticElements = false) {
+      var elems = new List<GSAElement>();
       var topoPts = new Point3dList(mesh.Vertices.ToPoint3dArray());
       var topoInts = new List<List<int>>();
       var ngons = mesh.GetNgonAndFacesEnumerable().ToList();
 
       foreach (MeshNgon ngon in ngons) {
-        var elem = new Element();
+
+        GSAElement elem = isLoadPanel ? new GSAElement(new LoadPanelElement()) : new GSAElement(new Element());
```
Comment (DominikaLos) [inline GsaGH/Helpers/GH/RhinoConversions.cs:491]: var here pls

Comment (SandeepArup) [reply GsaGH/Helpers/GH/RhinoConversions.cs:491]: image

**Review by DominikaLos — APPROVED** [review]

File: GsaGH/Helpers/GsaAPI/GsaElement.cs [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+        else {
+          return Element.OrientationNode;
+        }
+      }
+      set {
+        if (!IsLoadPanel) {
+          Element.OrientationNode = value;
+        }
+        else { throw new ArgumentException("Orientation angle is not applicable for load panels"); }
+      }
```
Comment (RaviKumarGubbala) [inline GsaGH/Helpers/GsaAPI/GsaElement.cs:176]: The exception should say OrientationNode (not "Orientation angle")

**Comment by RaviKumarGubbala** [comment]
NOTE: As discussed with @SandeepArup, 10.2.10 release is a blocker for this to be merged (as it requires the 10.2.10 GSA installer to compile the API code corresponding to Load Panels. CC: @a-tiwari17

File: GsaGH/Helpers/GsaAPI/GsaElement.cs (hunk 4 above)
Comment (SandeepArup) [reply GsaGH/Helpers/GsaAPI/GsaElement.cs]: @RaviKumarGubbala done

**Pull request merged**